### PR TITLE
Fix Test, Add Github Worflow, Fix Command builder.

### PR DESCRIPTION
**Pull request by Abhilash-Chandran**
The main changes are dispersed over 4 files. Most of the commits are local commits. If you prefer single commit for each file, I will recreate a new pull request.

1. `.github\workflow\build.yaml`
    * Adds a github workflow which builds, tests and update code coverage in all three OS.
2. `command_builder.dart`
    * The widget was returning same `onData` widget for all cases. Corrected to return expected widgets.
3. `flutter_command.dart`
    * Makes CommandError == able, and fixes toString for CommandResult.
    * Fixes the issue with restriction evaluation(from twitter discussion).
4. `flutter_command_test.dart`
    * Adds Collector class that holds a list of specific entries for comparisons purpose.
    * A CustomException was introduced to make == able, exception.
    * Mainly fixes existing AsyncCommand Test cases.
    * Adapts SyncCommand tests to validate collectors.
    * adds tests for `CommandBuilder` widget
    * Adds tests for global parameters like `globalExceptionHandler`, `loggingHandler`, `catchAlwaysDefault`.
    * Adds tests for MockCommand.

We can discuss the changes if you need clarity. It would be nice if we could verify that all the scenarios are covered. While code coverage shows nearly everything is covered, logical completeness needs to be asserted and more tests can be added. 💯 😃 

_**Documentation Change**_: fixes the error_handling flow diagram.

> PS: Next step Docusaurus 🤯 😄..!

### CODE REVIEW

File: test/flutter_command_test.dart [hunk 1]
```diff
@@ -1,561 +1,1147 @@
...
+      expect(command.thrownExceptions.value, null);
+      expect(executionCount, 1);
+
+      expect(command.canExecute.value, true);
+
+      // Verify Collectors
+      expect(thrownExceptionCollector.values, isNull);
+      expect(
+          cmdResultCollector.values,
+          containsAllInOrder([
```
Comment (escamoteur) [inline test/flutter_command_test.dart]: Why do you use `containsAllinOrder` when you could just pass the list as expected value?

**Review by escamoteur — APPROVED** [review]
Great work! 
only one question, in some tests you use `containsAllInOrder` do compare lists, in others you just pass a list as expected value. Any reason why you choose that way?

File: test/flutter_command_test.dart (hunk 1 above)
Comment (Abhilash-Chandran) [reply test/flutter_command_test.dart]: Actually no reason in these cases, I just left them there. containsAllIInOrder allows the main list to be interleaved which I thought could be useful later but actually I ended up passing all the expected values. I can remove it and replace them with normal list.

Comment (escamoteur) [reply test/flutter_command_test.dart]: I think that would be better, thanks!

**Pull request merged**